dotnet/src/Microsoft.AutoGen/Agents/HostBuilderExtensions.cs

```diff
@@ -71,7 +71,52 @@ public static AgentApplicationBuilder AddAgentWorker(this IHostApplicationBuilde
                                                   .Where(i => i.IsGenericType && i.GetGenericTypeDefinition() == typeof(IHandle<>))
                                                   .Select(i => (GetMessageDescriptor(i.GetGenericArguments().First())?.FullName ?? "")).ToHashSet()))
                                     .ToDictionary(item => item.t, item => item.Item2);
+            // if the assembly contains any interfaces of type IHandler, then add all the methods of the interface to the eventsMap
+            var handlersMap = AppDomain.CurrentDomain.GetAssemblies()
```

Comment:
That grabs all `AgentBase` implementations. What if user just wants to register partial of agents.

Reply:
There are alternate signatures / builder methods for adding a single agent or a user defined set of AgentTypes.